Kata/DiscountCalculator.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Linq;
+
+namespace Kata
+{
+    public class DiscountCalculator
+    {
+        public AfterTaxDiscounter afterTaxDiscounter { get; private set; }
+        public BeforeTaxDiscounter beforeTaxDiscounter { get; private set; }
+        public DiscountCalculator(ListOfDiscountsWithDetails discounts)
+        {
+            GenerateDiscountCalculators(discounts);
+        }
+
+        private void GenerateDiscountCalculators(ListOfDiscountsWithDetails discounts)
+        {
+            var before = discounts.ListOfDiscounts.Where(disc => disc.Precedence == DiscountPrecedence.Before).ToList();
+            var after = discounts.ListOfDiscounts.Where(disc => disc.Precedence == DiscountPrecedence.After).ToList();
+            if (after == null)
+                afterTaxDiscounter = null;
+            else
+            {
+                if (after.Count == 2)
+                    afterTaxDiscounter = new AfterTaxDiscounter(new AfterTaxDefaultDiscounter(), new AfterTaxUPCDiscounter());
```

Comment:
I feel like we should always pass both parameters, having optional params indicates a week class that good have mixed responsibilities, I would think of a way to move in the logic of `if (after.Count == 2)` and `if (after[0].DiscountType == DiscountType.Default)` to `AfterTaxDiscounter` class itself.